packages/cli/src/oclif/commands/test.js

```diff
@@ -0,0 +1,89 @@
+const { flags } = require('@oclif/command');
+
+const BaseCommand = require('../ZapierBaseCommand');
+const ValidateCommand = require('./validate');
+const constants = require('../../constants');
+const utils = require('../../utils');
+const { buildFlags } = require('../buildFlags');
+
+class TestCommand extends BaseCommand {
+  async perform() {
+    if (!this.flags['skip-validate']) {
+      await ValidateCommand.run([]);
+      if (process.exitCode) {
+        return;
+      }
+    }
+
+    const extraEnv = {
+      ZAPIER_BASE_ENDPOINT: constants.BASE_ENDPOINT
+    };
+
+    if (this.debug.enabled) {
+      extraEnv.LOG_TO_STDOUT = 'true';
+      extraEnv.DETAILED_LOG_TO_STDOUT = 'true';
+    }
+
+    const credentials = await utils.readCredentials(false);
+    if (credentials.deployKey) {
+      this.log(
+        `Adding ${constants.AUTH_LOCATION} to environment as ZAPIER_DEPLOY_KEY...`
+      );
+      extraEnv.ZAPIER_DEPLOY_KEY = credentials.deployKey;
+    }
+
+    const env = Object.assign({}, process.env, extraEnv);
+    const argv = [
+      'run',
+      '--silent',
+      'test',
+      '--',
+      `--timeout=${this.flags.timeout}`
+    ];
+    if (this.flags.grep) {
+      argv.push(`--grep=${this.flags.grep}`);
+    }
+
+    this.log('Running test suite.');
+
+    const packageManager = this.flags.yarn ? 'yarn' : 'npm';
+    const output = await utils.runCommand(packageManager, argv, {
+      stdio: 'inherit',
+      env
+    });
+    if (output.stdout) {
+      this.log(output.stdout);
+    }
+  }
+}
+
+TestCommand.flags = buildFlags({
+  commandFlags: {
+    timeout: flags.integer({
+      char: 't',
+      description: 'Set test-case timeout in milliseconds',
+      default: 2000
+    }),
+    grep: flags.string({
+      description: 'Only run tests matching pattern'
+    }),
+    'skip-validate': flags.boolean({
+      description: 'Forgo running `zapier validate` before `npm test`'
+    }),
+    yarn: flags.boolean({
+      description: 'Use yarn instead of npm'
+    })
+  }
+});
+
+TestCommand.args = [];
+
+TestCommand.examples = [
+  'zapier test',
+  'zapier test -t 30000 --grep api --skip-validate'
+];
+TestCommand.description = `Tests your app via \`npm test\`.
+
+This command is effectively the same as \`npm test\`, except we also validate your app and set up the environment. We recommend using mocha as your testing framework.`;
```

Comment:
Mocha is decent, but i wonder if we don't be a little more framework-neutral and not push them towards anything specific here. 

Also, rather than support specific flags to pass into any particular framework, we could pass along the rest of argv directly to npm/yarn. that way, we won't get anyone writing in saying "Could you add support for the `--xyz` flag to `zapier test`?" i'm down to mention these potentially helpful args in the docstring though!

Reply:
I like your idea! But I feel this is out of the scope of this PR. Maybe we should do it as seperate work?

Reply:
that's fine with me! the only reservation would be releasing the current items as CLI 9 and then having to do a second breaking change release soonish after. I'd rather do all the breaks together. 

In terms of this PR, i'm happy to see it merged as long as we take a holistic look at the args before we cut a release 👍

Reply:
I agree it's better to do all the breaks together. So yeah, we should do an overall review before we release 9.0.0.